docs/agent_api.rst

```diff
@@ -3,4 +3,313 @@
+
+------
+
+.. code-block:: cpp
+
+    bool run();
+
+Starts the execution of the previously created agent, until finished via an user interrupt or a process error.
+
+------
+
+.. code-block:: cpp
+
+    template <typename ... Args>
+    void add_middleware_callback(const Middleware::Kind& middleware_kind, const middleware::CallbackKind& callback_kind, std::function<void (Args ...)>&& callback_function);
+
+Sets an user-defined callback function for a specific create/delete middleware entity operation.
+
+:middleware_kind: Enum defining all the supported middlewares (see :ref:`middleware_abstraction_layer`).
+:callback_kind: Enum holding all the possible create/delete operations:
+
+.. code-block:: cpp
+
+    enum class CallbackKind : uint8_t
+    {
+        CREATE_PARTICIPANT,
+        CREATE_DATAWRITER,
+        CREATE_DATAREADER,
+        CREATE_REQUESTER,
+        CREATE_REPLIER,
+        DELETE_PARTICIPANT,
+        DELETE_DATAWRITER,
+        DELETE_DATAREADER,
+        DELETE_REQUESTER,
+        DELETE_REPLIER
+    };
+
+:callback_function: Callback to be defined by the user. It must follow a certain signature, depending on the used middleware.
+
+------
+
+.. _agent_class_api:
+
+eprosima::uxr::Agent
+^^^^^^^^^^^^^^^^^^^^
+
+However, it is also possible for users to create and instantiate their own agent instances, for example, to implement a :ref:`custom_transport_agent`.
+Also, in some scenarios it could be useful to have all the necessary `ProxyClient` and their associated DDS entities created by the agent even before
+clients are started, so that client applications get rid of the process of creating the session and the DDS entities and can focus only on the communication.
+
+This would allow a Micro XRCE-DDS Client application to be as tiny in terms of memory consumption as it can be.
+
+The following API is provided to fulfill these requirements:
+
+------
+
+.. code-block:: cpp
+
+    bool create_client(uint32_t key, uint8_t session, uint16_t mtu, Middleware::Kind middleware_kind, OpResult& op_result);
+
+Allows to create a `ProxyClient` entity, which can act on behalf of an external client to request the creation/deletion of DDS entities.
+
+Returns `true` if the creation was successful, or `false` otherwise.
+
+:key: The `ProxyClient`'s identifier.
+:session: The session ID to which the created `ProxyClient` is attached to.
+:mtu: The *Maximum Transmission Unit* size.
+:middleware_kind: The middleware used by the `ProxyClient`, to be chosen among the ones presented in the :ref:`middleware_abstraction_layer`.
+:op_result: The result status of this operation.
+
+------
+
+.. code-block:: cpp
+
+    bool delete_client(uint32_t key, OpResult& op_result);
+
+Deletes a certain `ProxyClient` from the client proxy database, given its ID.
+
+Returns `true` if the operation was completed successfully, or `false` otherwise (for example, if the given ID was not registered to any `ProxyClient`).
```

Comment:
```suggestion
Returns ``true`` if the operation was completed successfully, ``false`` otherwise (for example, if the provided ID was not registered to any ``ProxyClient``).
```